src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -245,40 +246,83 @@ private static String getInstanceName(DetailAST methodCallAst) {
             DetailAST variableUsageAst, String variableName) {
         boolean result = true;
         boolean isUsedVariableDeclarationFound = false;
         DetailAST currentSiblingAst = variableUsageAst;
         String initInstanceName = "";
+        boolean isVarNotInInitializationSequence = true;
 
         while (result && !isUsedVariableDeclarationFound && currentSiblingAst != null) {
-            if (currentSiblingAst.getType() == TokenTypes.EXPR
-                    && currentSiblingAst.getFirstChild().getType() == TokenTypes.METHOD_CALL) {
-                final DetailAST methodCallAst = currentSiblingAst.getFirstChild();
-                final String instanceName = getInstanceName(methodCallAst);
-                if (instanceName.isEmpty()) {
-                    result = false;
-                }
-                else if (!instanceName.equals(initInstanceName)) {
-                    if (initInstanceName.isEmpty()) {
-                        initInstanceName = instanceName;
+            switch (currentSiblingAst.getType()) {
+                case TokenTypes.EXPR:
+                    if (currentSiblingAst.getFirstChild().getType() == TokenTypes.METHOD_CALL) {
+                        final DetailAST methodCallAst = currentSiblingAst.getFirstChild();
+                        final String instanceName = getInstanceName(methodCallAst);
+                        if (initInstanceName.isEmpty()) {
+                            initInstanceName = instanceName;
+                        }
+                        isVarNotInInitializationSequence = isVarNotInInitializationSequence
+                            && !isVariableUsed(variableName, methodCallAst);
+                        result = areInstanceNamesEqual(initInstanceName, instanceName);
                     }
                     else {
                         result = false;
                     }
-                }
+                    break;
+                case TokenTypes.VARIABLE_DEF:
+                    final String currentVariableName =
+                            currentSiblingAst.findFirstToken(TokenTypes.IDENT).getText();
+                    isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
+                    break;
 
+                default:
+                    result = currentSiblingAst.getType() == TokenTypes.SEMI;
+                    break;
             }
-            else if (currentSiblingAst.getType() == TokenTypes.VARIABLE_DEF) {
-                final String currentVariableName =
-                        currentSiblingAst.findFirstToken(TokenTypes.IDENT).getText();
-                isUsedVariableDeclarationFound = variableName.equals(currentVariableName);
+            currentSiblingAst = currentSiblingAst.getPreviousSibling();
+        }
+        boolean ans = result;
+        if (variableUsageAst != null) {
+            ans = result && !isVarNotInInitializationSequence;
+        }
+        return ans;
```

Comment:
Please refactor this to not have mutable variables and use more descriptive names.

Reply:
done 
I have make it simple.